### ISSUE TITLE
PHP `rename` method doesn't work for ramdisk in Linux

### ISSUE THREAD

**Issue by kohenkatz** [phpbrew/phpbrew#669]
I followed the [instructions in the Wiki](https://github.com/phpbrew/phpbrew/wiki/Cookbook#using-ramdisk-to-build-php) to build PHP on a RAM-disk, and I got the following error (full output):

```
===> phpbrew will now build 7.0.2
===> Loading and resolving variants...
Checking distribution checksum...
Checksum matched: 8c71b2ffd29311e4b7ac7788bd107554
===> Extracting /opt/phpbrew/distfiles/php-7.0.2.tar.bz2 to /opt/phpbrew/tmp/php-7.0.2
===> Moving /opt/phpbrew/tmp/php-7.0.2 to /tmp/ramdisk/php-7.0.2
PHP Warning:  rename(): The first argument to copy() function cannot be a directory in phar:///usr/bin/phpbrew/src/PhpBrew/Tasks/ExtractTask.php on line 59
PHP Warning:  rename(/opt/phpbrew/tmp/php-7.0.2,/tmp/ramdisk/php-7.0.2): Invalid cross-device link in phar:///usr/bin/phpbrew/src/PhpBrew/Tasks/ExtractTask.php on line 59
RuntimeException: Unable to move /opt/phpbrew/tmp/php-7.0.2 to /tmp/ramdisk/php-7.0.2
Trace:

Trace:

    0) PhpBrew\Tasks\ExtractTask->extract(PhpBrew\Build, '/opt/phpbrew/distfiles/php-7.0.2.tar.bz2', /tmp/ramdisk)
    1) PhpBrew\Command\InstallCommand->execute('7.0.2', '+default', '+dbs', '+cgi', '+exif', '+fpm', '+gd', '+gettext', '+hash', '+iconv', '+imap', '+intl', '+ipv6', '+openssl', '+opcache')
    2) call_user_func_array([PhpBrew\Command\InstallCommand, 'execute'], ['7.0.2', '+default', '+dbs', '+cgi', '+exif', '+fpm', '+gd', '+gettext', '+hash', '+iconv', '+imap', '+intl', '+ipv6', '+openssl', '+opcache'])
    3) CLIFramework\CommandBase->executeWrapper(['7.0.2', '+default', '+dbs', '+cgi', '+exif', '+fpm', '+gd', '+gettext', '+hash', '+iconv', '+imap', '+intl', '+ipv6', '+openssl', '+opcache'])
    4) CLIFramework\Application->run(['/usr/bin/phpbrew', 'install', '--build-dir', '/tmp/ramdisk/', '7.0.2', '+default', '+dbs', '+cgi', '+exif', '+fpm', '+gd', '+gettext', '+hash', '+iconv', '+imap', '+intl', '+ipv6', '+openssl', '+opcache'])
    5) CLIFramework\Application->runWithTry(['/usr/bin/phpbrew', 'install', '--build-dir', '/tmp/ramdisk/', '7.0.2', '+default', '+dbs', '+cgi', '+exif', '+fpm', '+gd', '+gettext', '+hash', '+iconv', '+imap', '+intl', '+ipv6', '+openssl', '+opcache'])
    6) require('phar:///usr/bin/phpbrew/scripts/phpbrew-emb.php')
```

Here is the command I used:

```
phpbrew install --build-dir /tmp/ramdisk/ 7.0.2 +default +dbs +cgi +exif +fpm +gd +gettext +hash +iconv +imap +intl +ipv6 +openssl +opcache
```

The problem is that, at least on Linux, the `rename` method only works cross-filesystem on **files**, not on **directories**.

This is reported as [PHP Bug 54097](https://bugs.php.net/bug.php?id=54097), which is 5 years old with no fix, so it's likely that the behavior will not change.

If PHPBrew is going to keep the option to use a RAM-disk, this should really be fixed.

**Comment by c9s** [comment]
looks like `rename` calls `copy` on ramdisk

**Comment by c9s** [comment]
I agree we should avoid that

**Comment by kohenkatz** [comment]
@c9s `rename` calls `copy` on *all* cross-device usage (at least on Linux), not just on ramdisks.

**Comment by c9s** [comment]
I am not sure why we were renaming the extracted directory to another instead of *just extracting` to the directory.  @mdkholy do you remember the reason that why we need to rename the directory?

**Comment by c9s** [comment]
Now I recall that - we always extract to a temp directory because the tarball file sometimes may be corrupted.

The solution will be - Extract the tarball file to the target build directory.

**Comment by c9s** [comment]
Fixed in 3da4161b07a9cf56d506812ee63d69a34e2ebb58...dbb33a24fe41f464c067b024c988436f14953ac1
